barman/utils.py

```diff
@@ -902,3 +902,39 @@ def lock_files_cleanup(lock_dir, lock_directory_cleanup):
                 _logger.debug(
                     "%s file lock already acquired, skipping removal" % filename
                 )
+
+
+def edit_config(file, section, option, value):
+    """
+    Utility method that given a file and a config section allows to:
+        - add a new section if at least a key-value content is provided
+        - add a new key-value to a config section
+        - change a section value
+
+    :param file: the path to the file to edit
+    :param section: the config section to edit or to add
+    :param option: the config key to edit or add
+    :param value: the value for the config key to update or add
+    """
+    conf_section = False
+    idx = 0
+    with open(file, "r") as config:
+        lines = config.readlines()
+        eof = len(lines) - 1
+        for idx, line in enumerate(lines):
+            if conf_section and line.startswith("["):  # next section
+                lines.insert(idx - 1, option + " = " + value)
+                break
+            elif conf_section and line.startswith(option + " = "):
```

Comment:
```suggestion
            elif conf_section and line.replace(" ", "").startswith(option + "="):
```

Remove all spaces, just in case something weird like this comes up:

```python
line = "   some_option          =some_value"
```

Reply:
What happens if `some_value` has valid white spaces in it?

Reply:
Oh, this is only in the `if` statement, and only for the parameter name.
We are interested in checking if the line "starts" with `OPTION_NAME=`, `OPTION_NAME` being the setting that we want to change the value.